plugins/inputs/logfile/logfile.go

```diff
@@ -259,6 +266,10 @@ func (t *LogFile) FindLogSrc() []logs.LogSrc {
 		}
 	}
 
+	if allowedOpenFileLimit, err := fdlimit.CurrentOpenFileLimit(); err == nil {
```

Comment:
We should not keep looking up this file descriptor limit. We can get this information from the buffered channel and only look up the file descriptor limit once. 

`len(ch)` to see how many things are currently in the channel, and `cap(ch)` to see the allowed capacity for the channel.

Reply:
Agree. I have not known in go channels we could use `cap(ch)`. First time dealing with `go channel`